test/case/infix_interfaces/bridge_vlan_separation.py

```diff
@@ -0,0 +1,171 @@
+                            "vlans": {
+                                "vlan": [
+                                    {
+                                        "vid": 10,
+                                        "untagged": [ tport20 ],
+                                        "tagged":   [ "br0", tport22 ]
+                                    },
+                                    {
+                                        "vid": 20,
+                                        "untagged": [ tport21 ],
+                                        "tagged":   [ "br0", tport22 ]
+                                    }
+                                ]
+                            }
+                        }
+                    },
+                    {
+                        "name": tport20,
+                        "type": "iana-if-type:ethernetCsmacd",
+                        "enabled": True,
+                        "infix-interfaces:bridge-port": {
+                            "pvid": 10,
+                            "bridge": "br0"
+                        }
+                    },
+                    {
+                        "name": tport21,
+                        "type": "iana-if-type:ethernetCsmacd",
+                        "enabled": True,
+                        "infix-interfaces:bridge-port": {
+                            "pvid": 20,
+                            "bridge": "br0"
+                        }
+                    },
+                    {
+                        "name": tport22,
+                        "type": "iana-if-type:ethernetCsmacd",
+                        "enabled": True,
+                        "infix-interfaces:bridge-port": {
+                            "bridge": "br0",
+                        }
+                    }
+                ]
+            }
+        })
+
+    with test.step("Ping host:data20 with IP 10.0.0.3 and host:data21 with IP 10.0.0.4 from host:data11 with IP 10.0.0.2"):
+        
+        _, hport10 = env.ltop.xlate("host", "data10")
+        _, hport11 = env.ltop.xlate("host", "data11")
+        _, hport20 = env.ltop.xlate("host", "data20")
+        _, hport21 = env.ltop.xlate("host", "data21")
+
+        with infamy.IsolatedMacVlan(hport10) as ns10, \
+             infamy.IsolatedMacVlan(hport11) as ns11, \
+             infamy.IsolatedMacVlan(hport20) as ns20, \
+             infamy.IsolatedMacVlan(hport21) as ns21:
+
+            ns10.addip("10.0.0.1")
+            ns11.addip("10.0.0.2")
+            ns20.addip("10.0.0.3")
+            ns21.addip("10.0.0.4")
+
+            ns10.ping("10.0.0.3")
+            ns11.ping("10.0.0.4")
+
+            if not ns10.ping_neg("10.0.0.4"):
+                test.fail()
+            if not ns11.ping_neg("10.0.0.3"):
+                test.fail()
+            if not ns10.ping_neg("10.0.0.2"):
+                test.fail()
+            if not ns11.ping_neg("10.0.0.1"):
+                test.fail()
+                    
+    test.succeed()
```

Comment:
For clarity, I think we should have separate IP subnets for the two VLANs used in this test.  Even though they are VLAN separated in the switch and isolated macvlan interfaces on the host, it's confusing when reading the test.  E.g.,

 - 10.0.10.0/24, for hosts ns10 and ns20
 - 10.0.20.0/24, for hosts ns11 and ns21

What do you think, @ahmkar94 and @wkz?

Reply:
I agree it looks much cleaner with the separate subnets.
Somehow I had a feeling that the test is "stronger" and I am really testing bridge configuration if all the IPs are in the same subnet.

The way the test is implemented I wouldn't be able to know whether I don't get ping reply due to misconfiguration or due to separate subnets? Or...

Reply:
You are right, of course, and @wkz concurs ... just keep it as-is, I'm happy.  Just wait for @wkz to review.